neurom/apps/morph_stats.py

```diff
@@ -73,29 +74,30 @@ def _stat_name(feat_name, stat_mode):
 
 def extract_stats(neurons, config):
     '''Extract stats from neurons'''
-
     stats = defaultdict(dict)
-    for ns, modes in config['neurite'].items():
+
+    def _fill_compoundified(data, stat_name, stat):
+        '''Insert the stat in the dict and eventually split it into XYZ components'''
+        if stat is None or not isinstance(stat, np.ndarray) or stat.shape not in ((3, ), ):
+            data[stat_name] = stat
+        else:
+            for i, suffix in enumerate('XYZ'):
+                compound_stat_name = stat_name + '_' + suffix
+                data[compound_stat_name] = stat[i]
+
+    for feature_name, modes in config['neurite'].items():
         for n in config['neurite_type']:
```

Comment:
Can we use here a cartesian product like `itertools.product` for `modes` and ` config['neurite_type']` ?

Reply:
@wizmer this one has been overlooked.

Reply:
Sorry for that. Fixed.